### PR TITLE
Fio domain registration

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: src/components/scenes/FioNameConfirmScene.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  blockPadding: {
+    paddingTop: THEME.rem(4),
+    paddingLeft: THEME.rem(1.25),
+    paddingRight: THEME.rem(1.25)
+  },
+  spacer: {
+    paddingTop: THEME.rem(1.25)
+  }
+}
+const styles: typeof rawStyles = StyleSheet.create(rawStyles)
```
Comment (andreyvEze) [inline src/components/scenes/FioNameConfirmScene.js:255]: change to be THEMEd, transaction details like as example

File: ios/Podfile.lock [hunk 2]
```diff
@@ -334,4 +334,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 08d7925f21ad0118fdb477099b500ec9a735726d
 
-COCOAPODS: 1.9.3
+COCOAPODS: 1.10.0.beta.2
```
Comment (andreyvEze) [inline ios/Podfile.lock:337]: remove

File: src/components/scenes/FioAddressListScene.js [hunk 3]
```diff
@@ -185,7 +170,8 @@ const rawStyles = {
     marginLeft: scale(5)
   },
   button: {
-    padding: scale(10)
+    paddingVertical: scale(5),
+    paddingHorizontal: scale(10)
```
Comment (andreyvEze) [inline src/components/scenes/FioAddressListScene.js:174]: THEME.rem

File: src/components/scenes/FioAddressRegisterScene.js [hunk 4]
```diff
@@ -480,6 +480,13 @@ const rawStyles = {
     paddingLeft: THEME.rem(0.75),
     paddingRight: THEME.rem(0.75),
     paddingVertical: THEME.rem(0.75)
+  },
+  iconIon: {
```
Comment (andreyvEze) [inline src/components/scenes/FioAddressRegisterScene.js:484]: remove it

File: src/components/scenes/FioAddressRegisterSelectWalletScene.js [hunk 5]
```diff
@@ -353,6 +353,13 @@ const rawStyles = {
     paddingLeft: THEME.rem(0.75),
     paddingRight: THEME.rem(0.75),
     paddingVertical: THEME.rem(0.75)
+  },
+  iconIon: {
```
Comment (andreyvEze) [inline src/components/scenes/FioAddressRegisterSelectWalletScene.js:357]: remove

File: src/components/scenes/FioDomainRegisterScene.js [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+  isConnected: boolean
+}
+
+export type DispatchProps = {
+  createFioWallet: () => Promise<EdgeCurrencyWallet>
+}
+
+type Props = StateProps & DispatchProps & ThemeProps
+
+const ionIconSize = THEME.rem(4)
```
Comment (andreyvEze) [inline src/components/scenes/FioDomainRegisterScene.js:53]: remove, use from { theme } = this.props

File: src/components/scenes/FioDomainRegisterScene.js [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+        width: '100%'
+      }
+    }
+
+    return (
+      <SafeAreaView>
+        <Gradient style={styles.scrollableGradient} />
+        <ScrollView ref="_scrollView">
+          <View style={styles.scrollableView}>
+            <IonIcon name="ios-at" style={styles.iconIon} color={THEME.COLORS.BLUE_3} size={ionIconSize} />
```
Comment (andreyvEze) [inline src/components/scenes/FioDomainRegisterScene.js:267]: theme from props

File: src/components/scenes/FioDomainRegisterScene.js [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+                  style={materialInputOnWhiteStyle}
+                  clearButtonMode={this.clearButtonMode}
+                  autoCorrect={false}
+                  autoCapitalize="none"
+                  placeholder={s.strings.fio_domain_label}
+                  caretHidden
+                  onFocus={this.handleFioDomainFocus}
+                  onChangeText={this.handleFioDomainChange}
+                  onSubmitEditing={this.handleNextButton}
+                  selectionColor={THEME.COLORS.ACCENT_MINT}
```
Comment (andreyvEze) [inline src/components/scenes/FioDomainRegisterScene.js:287]: use from props

File: src/locales/en_US.js [hunk 9]
```diff
@@ -587,12 +589,16 @@ const strings = {
   fio_address_features: '• Works across all tokens and coins\n• Secured from your wallet\n• Completely decentralized',
   fio_address_first_screen_end:
     'Similar to website names, your FIO Address can be used to send you tokens/coins without ever having to see or send your public key.',
-  fio_address_reg_domain_label: 'Want to register your own FIO Domain such as @mydomain?',
   fio_address_reg_domain: 'Register FIO Domain',
+  fio_domain_reg_text: 'Need to note that you need to use FIO to register an address on a custom domain.',
```
Comment (andreyvEze) [inline src/locales/en_US.js:593]: Note that you need to use FIO...

**Comment by andreyvEze** [comment]
approved

File: src/components/scenes/FioDomainRegisterScene.js [hunk 10]
```diff
@@ -0,0 +1,483 @@
...
+  isConnected: boolean
+}
+
+export type DispatchProps = {
+  createFioWallet: () => Promise<EdgeCurrencyWallet>
+}
+
+type Props = StateProps & DispatchProps & ThemeProps
+
+class FioDomainRegister extends React.Component<Props, LocalState> {
```
Comment (paullinator) [inline src/components/scenes/FioDomainRegisterScene.js:51]: Please use React.PureComponent

File: src/components/scenes/FioDomainRegisterSelectWalletScene.js [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+          },
+          dismissAlert: true,
+          lockInputs: true,
+          onDone: (error: Error | null, edgeTransaction?: EdgeTransaction) => {
+            if (error) {
+              setTimeout(() => {
+                showError(s.strings.create_wallet_account_error_sending_transaction)
+              }, 750)
+            } else if (edgeTransaction) {
+              Alert.alert(
```
Comment (paullinator) [inline src/components/scenes/FioDomainRegisterSelectWalletScene.js:140]: Do not use the system alert. Use an airship modal instead. ButtonModal is ideal and easy to use.

File: src/components/scenes/FioNameConfirmScene.js [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+              publicKey: ownerPublicKey
+            },
+            true
+          )
+          if (response.error) {
+            if (response.error === 'You have already registered a free address for that domain' && response.code === 400) {
+              const publicDomains = await fioPlugin.otherMethods.getDomains(fioPlugin.currencyInfo.defaultSettings.fallbackRef)
+              const domainExists = publicDomains.find(domain => domain.domain === fioName.split(Constants.FIO_ADDRESS_DELIMITER)[1])
+              if (domainExists && !domainExists.free) {
+                Alert.alert(s.strings.fio_address_register_pay_title, s.strings.fio_address_register_pay, [{ text: s.strings.string_ok_cap }])
```
Comment (paullinator) [inline src/components/scenes/FioNameConfirmScene.js:119]: Do not use the system alert. Use an airship modal instead. ButtonModal is ideal and easy to use.

File: src/components/scenes/FioNameConfirmScene.js [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+                    walletId: '',
+                    isFree: domainExists.free
+                  },
+                  isFallback: true
+                })
+              }
+            }
+            throw new Error(response.error)
+          }
+          Alert.alert(
```
Comment (paullinator) [inline src/components/scenes/FioNameConfirmScene.js:136]: Do not use the system alert. Use an airship modal instead. ButtonModal is ideal and easy to use.

File: src/components/scenes/FioDomainRegisterSelectWalletScene.js [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+type LocalState = {
+  loading: boolean,
+  supportedCurrencies: { [currencyCode: string]: boolean },
+  paymentInfo: { [currencyCode: string]: { amount: string, address: string } },
+  activationCost: number
+}
+
+type Props = NavigationProps & StateProps & DispatchProps & ThemeProps
+
+class FioDomainRegisterSelectWallet extends React.Component<Props, LocalState> {
```
Comment (paullinator) [inline src/components/scenes/FioDomainRegisterSelectWalletScene.js:55]: Please use React.PureComponent

File: src/components/scenes/FioNameConfirmScene.js [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+export type NavigationProps = {
+  fioName: string,
+  paymentWallet: EdgeCurrencyWallet,
+  fee: number,
+  ownerPublicKey: string
+}
+
+type Props = NavigationProps & StateProps & ThemeProps
+
+class FioNameConfirm extends React.Component<Props, LocalState> {
```
Comment (paullinator) [inline src/components/scenes/FioNameConfirmScene.js:43]: Please use React.PureComponent

**Review by paullinator — CHANGES_REQUESTED** [review]

**Review by paullinator — APPROVED** [review]

**Comment by paullinator** [comment]
Approved but cannot merge until the latest account-based repo is published and upgraded

**Pull request merged**